v4l2.go

```diff
@@ -311,6 +344,55 @@ func getFrameSize(fd uintptr, index uint32, code uint32) (frameSize FrameSize, e
 	return
 }
 
+func getFrameInterval(fd uintptr, index uint32, code uint32, width uint32, height uint32) (FrameRate, error) {
+	frmivalEnum := &v4l2_frmivalenum{
+		index:        index,
+		pixel_format: code,
+		width:        width,
+		height:       height,
+	}
+
+	if err := ioctl.Ioctl(fd, VIDIOC_ENUM_FRAMEINTERVALS, uintptr(unsafe.Pointer(frmivalEnum))); err != nil {
+		return FrameRate{}, err
+	}
+
+	switch frmivalEnum._type {
+
+	case V4L2_FRMIVAL_TYPE_DISCRETE:
+		discrete := &v4l2_fract{}
+		if err := binary.Read(bytes.NewBuffer(frmivalEnum.union[:]), NativeByteOrder, discrete); err != nil {
```

Comment:
@bazile-clyde This where I get errors if v4l2_fract members are not exported. Not entirely sure why.

Reply:
Ah, this makes sense. `binary.Read` is going to write into `v4l2_fract` and needs its fields to be exported to do so since its in a different package.